nibabel/gifti/parse_gifti_fast.py

```diff
@@ -108,9 +108,6 @@ def StartElementHandler(self, name, attrs):
             self.img = GiftiImage()
             if 'Version' in attrs:
                 self.img.version = attrs['Version']
-            if 'NumberOfDataArrays' in attrs:
-                self.img.numDA = int(attrs['NumberOfDataArrays'])
-                self.count_da = False
```

Comment:
I think `attrs['NumberOfDataArrays']` should at least be validated and throw a warning if the count doesn't match the actual number of data arrays. Something like:

    if 'NumberOfDataArrays' in attrs:
        self.expected_numda = int(attrs['NumberOfDataArrays'])